From 988dd3f3363d8ab4ee53f61e0eb5afc6646c9d4f Mon Sep 17 00:00:00 2001
From: Nikita Popov <npopov@redhat.com>
Date: Tue, 26 Sep 2023 13:06:29 +0200
Subject: [PATCH] disable myst parser

---
 clang/docs/conf.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/clang/docs/conf.py b/clang/docs/conf.py
index ca310026f53e..dfb74273b5b2 100644
--- a/clang/docs/conf.py
+++ b/clang/docs/conf.py
@@ -35,9 +35,6 @@ templates_path = ["_templates"]
 
 import sphinx
 
-if sphinx.version_info >= (3, 0):
-    extensions.append("myst_parser")
-
 # The encoding of source files.
 # source_encoding = 'utf-8-sig'
 
-- 
2.41.0

